src/st/water_effects.h

```diff
@@ -87,207 +93,202 @@ void EntityAlucardWaterEffect(Entity* self) {
+                if (sp10[0] < 14) {
+                    var_s7 =
+                        ((14 - sp10[0]) << 11) + (g_WaterXTbl[sp30 + 2] << 8) +
+                        (g_WaterXTbl[sp30 + 5] << 5);
+                } else if (sp10[1] < 14) {
+                    var_s7 =
+                        ((sp10[1] + 14) << 11) + (g_WaterXTbl[sp30 + 2] << 8) +
+                        (g_WaterXTbl[sp30 + 6] << 5);
                 } else {
                     var_s7 = g_WaterXTbl[sp30 + 2] << 8;
                 }
                 var_s1 = (var_s7 >> 8) & 7;
-                if (var_s1 == 0 || var_s1 == 7) {
+                if (!var_s1 || var_s1 == 7) {
                     var_s1 = (var_s7 >> 5) & 7;
-                    if (var_s1 == 0 || var_s1 == 7) {
+                    if (!var_s1 || var_s1 == 7) {
                         var_s1 = 0;
                     }
                 }
                 if (var_s1) {
                     for (i = 0; i < 8; i++) {
-                        var_s0 =
+                        tempEntity =
                             AllocEntity(&g_Entities[224], &g_Entities[256]);
-                        if (var_s0 == NULL) {
+                        if (tempEntity == NULL) {
                             break;
                         }
                         CreateEntityFromEntity(
-                            E_SIDE_WATER_SPLASH, player, var_s0);
-                        var_s0->params =
+                            E_SIDE_WATER_SPLASH, player, tempEntity);
+                        tempEntity->params =
                             (u16)g_WaterXTbl[sp30 + 7] + (var_s1 << 4) + i;
-                        var_s0->posY.i.hi += (var_s6 - var_s3);
-                        var_s0->zPriority = player->zPriority;
+                        tempEntity->posY.i.hi += var_s6 - var_s3;
+                        tempEntity->zPriority = player->zPriority;
                     }
                 } else {
-                    var_s0 = AllocEntity(&g_Entities[224], &g_Entities[256]);
-                    if (var_s0 != NULL) {
-                        CreateEntityFromEntity(E_SPLASH_WATER, player, var_s0);
+                    tempEntity =
+                        AllocEntity(&g_Entities[224], &g_Entities[256]);
+                    if (tempEntity != NULL) {
+                        CreateEntityFromEntity(
+                            E_SPLASH_WATER, player, tempEntity);
                         if (player->velocityY > FIX(4)) {
-                            var_s0->params = var_s7 + 1;
+                            tempEntity->params = var_s7 + 1;
                         } else {
-                            var_s0->params = var_s7 + 2;
+                            tempEntity->params = var_s7 + 2;
                         }
-                        var_s0->posY.i.hi += (var_s6 - var_s3);
-                        var_s0->params =
-                            var_s7; // this nullifies the above block!
```

Comment:
This looks like a bug I think? So might be nice to keep a comment here to that effect